Sources/Build/ManifestBuilder.swift

```diff
@@ -268,12 +285,147 @@ extension LLBuildManifestBuilder {
+    // SwiftPM will process targets in a topological order and “bubble-up” each target’s
+    // inter-module dependency graph to its dependees. First, SwiftPM will process B, and be
+    // able to plan its full build because it does not have any target dependencies. Then the
+    // driver is tasked with planning a build for A. SwiftPM will pass as input to the driver
+    // the module dependency graph of its target’s dependencies, in this case, just the
+    // dependency graph of B. The driver is then responsible for the necessary post-processing
+    // to merge the dependency graphs and plan the build for A, using artifacts of B as explicit
+    // inputs.
+    public func addTargetsToExplicitBuildManifest() throws {
+        // Sort the product targets in topological order in order to collect and "bubble up"
+        // their respective dependency graphs to the depending targets.
+        let nodes: [ResolvedTarget.Dependency] = plan.targetMap.keys.map { ResolvedTarget.Dependency.target($0, conditions: []) }
+        let allTargets = try! topologicalSort(nodes, successors: { $0.dependencies })
+
+        // Collect all targets' dependency graphs
+        var targetDepGraphMap : [ResolvedTarget: InterModuleDependencyGraph] = [:]
+
+        // Create commands for all target descriptions in the plan.
+        for dependency in allTargets.reversed() {
+            guard case .target(let target, _) = dependency else {
+                // TODO
+                fatalError("Product Dependencies not supported in Explicit Module Build Mode.")
+            }
+            guard let description = plan.targetMap[target] else {
+                fatalError("Expected description for target: \(target)")
+            }
+            switch description {
+                case .swift(let desc):
+                    createExplicitSwiftTargetCompileCommand(description: desc,
+                                                            targetDepGraphMap: &targetDepGraphMap)
+                case .clang(let desc):
+                    createClangCompileCommand(desc)
+            }
+        }
+    }
+
+    private func createExplicitSwiftTargetCompileCommand(
+        description: SwiftTargetBuildDescription,
+        targetDepGraphMap: inout [ResolvedTarget: InterModuleDependencyGraph]
+    ) {
+        // Inputs.
+        let inputs = computeSwiftCompileCmdInputs(description)
+
+        // Outputs.
+        let objectNodes = description.objects.map(Node.file)
+        let moduleNode = Node.file(description.moduleOutputPath)
+        let cmdOutputs = objectNodes + [moduleNode]
+
+        // Commands.
+        addExplicitBuildSwiftCmds(description, inputs: inputs,
+                                  objectNodes: objectNodes,
+                                  moduleNode: moduleNode,
+                                  targetDepGraphMap: &targetDepGraphMap)
+
+        addTargetCmd(description, cmdOutputs: cmdOutputs)
+        addModuleWrapCmd(description)
```

Comment:
Probably unrelated to your PR, but I thought the driver took care of the module-wrap command?